### PR TITLE
feat: As a user i want to use a placeholder for my co-proposer until he/she creates the account

### PR DESCRIPTION
**Pull request by jekabs-karklins**
## Description

As a user, i want to use a placeholder for my co-proposer until he/she creates the account

## Motivation and Context

We want to allow users to start creating the proposal before all his/her co-proposers are registered.
This implementation also addresses the issue with email aliases, where the proposal creator creates a proposal but specifies a different email, which the co-proposer uses to sign up because he/she will use the code to join the proposal and he/she can use any account for that.

Additionally, the redeem-code feature is quite generic, and redeeming the code essentially takes all that the placeholder user has (proposals/roles/assignments) and adds it to the user who redeems the code.

## How Has This Been Tested

Added E2E tests
Added unit tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-2984

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: apps/user-office-frontend/src/hooks/common/ButtonWithDialog.tsx [hunk 1]
```diff
@@ -27,19 +33,20 @@ function ButtonWithDialog(props: ButtonWithDialogProps) {
...
-              variant="outlined"
-              onClick={() => setIsDialogOpen(false)}
-              data-cy="close-dialog"
-            >
-              Close
-            </Button>
-          </ActionButtonContainer>
+          {React.cloneElement(children, {
+            onClose: () => setIsDialogOpen(false),
+          })}
```
Comment (Junjiequan) [inline apps/user-office-frontend/src/hooks/common/ButtonWithDialog.tsx:49]: Nice change - thumbs up

**Review by bashanlam — COMMENTED** [review]
Is it necessary to do proposal_user and redeem_codes record insertion (apps/user-office-backend/db_patches/db_seeds/0006_RedeemCodes.sql)?

File: .github/workflows/test-build.yml [hunk 2]
```diff
@@ -6,6 +6,9 @@ on:
   pull_request:
     branches: [develop]
 
+env:
+  NODE_VERSION: 16.x
+
```
Comment (mutambaraf) [inline .github/workflows/test-build.yml:11]: I assume fix to node has been done somewhere else. Happy to approve this.

**Review by martin-trajanovski — APPROVED** [review]
There is a small comment but otherwise it looks good to me 💯

**Pull request merged**